CourseManager/Courses/Course 1 Build a SwiftUI app for iOS 15/Model/Coin.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  Coin.swift
+//  CourseManager
+//
+//  Created by Summer Crow on 2022-11-11.
+//
+
+import Foundation
+
+struct Coin: Identifiable, Decodable {
+    var id: Int
+    var coin_name: String
+    var acronym: String
+    var logo: String
+}
+
+class CoinModel: ObservableObject {
+    @Published var coins: [Coin] = []
```

Comment:
It is better to make them `private(set)` so nobody can't modify them from the View.

Reply:
Wasn't aware of this, good to know. I've updated but also made the fetch func private as follows. Is this a good way to go about it?

 @Published private(set) var coins: [Coin] = []
   init() {
        Task(priority: .high) {
            await fetchCoins()
        }
    }
    @MainActor
    private func fetchCoins() async {
        do {
            guard let url = URL(string: "https://random-data-api.com/api/crypto_coin/random_crypto_coin?size=10") else {
                return
            }
            let (data, _) = try await URLSession.shared.data(from: url)
            coins = try JSONDecoder().decode([Coin].self, from: data)
        } catch(let error) {
            print(error)
        }
    }